libraries/stdlib/osl/stdosl.h

```diff
@@ -573,6 +573,169 @@ closure color volume_matte() BUILTIN;
+​
+// Constructs an EDF emitting light uniformly in all directions.
+//
+closure color uniform_edf(color emittance) BUILTIN;
+
+// Constructs an EDF emitting light inside a cone around the normal direction.
+//
+closure color conical_edf(color emittance, normal N, float inner_angle, float outer_angle) BUILTIN;
+
+// Constructs an EDF emitting light according to a measured IES light profile.
+//
+closure color measured_edf(color emittance, normal N, string file) BUILTIN;
+​
+​
+// -------------------------------------------------------------//
+// VDF closures                                                 //
+// -------------------------------------------------------------//
+​
+// Constructs a VDF for pure volumetric absorption.
+//
+closure color absorption_vdf(color absorption) BUILTIN;
+​
+// Constructs a VDF scattering light for a participating medium, based on the Henyey-Greenstein
+// phase function. Forward, backward and uniform scattering is supported and controlled by the
+// anisotropy input.
+//
+closure color anisotropic_vdf(color absorption, color scattering, float anisotropy) BUILTIN;
+​
+​
+// -------------------------------------------------------------//
+// Layering closures                                            //
+// -------------------------------------------------------------//
+​
+// Vertically layer a layerable BSDF such as dielectric_bsdf, generalized_schlick_bsdf or
+// sheen_bsdf over a BSDF or VDF. The implementation is target specific, but a standard way
+// of handling this is by albedo scaling, using "base*(1-reflectance(top)) + top", where
+// reflectance() calculates the directional albedo of a given top BSDF.
+//
+// TODO:
+// - This could also be achived by closure nesting where each layerable closure takes
+//   a closure color "base" input instead.
+// - One advantage having a dedicated layer() closure is that in the future we may want to
+//   introduce parameters to describe the sandwitched medium between the layer interfaces.
+//   Such parameterization could then be added on this layer() closure as extra arguments.
+// - Do we want/need parameters for the medium here now, or do we look at that later?
+//
+closure color layer(closure color top, closure color base) BUILTIN;
+​
+// Mix two closures according to a weight. Performs horizontal layering by linear interpolation
+// between the two inputs: "bg∗(1−weight) + fg∗weight".
+//
+closure color mix(closure color bg, closure color fg, float weight);
+​
+​
+// -------------------------------------------------------------//
+// Utility functions                                            //
+// -------------------------------------------------------------//
+​
+// Converts the artistic parameterization reflectivity and edge_color to complex IOR values.
+// To be used with the conductor_bsdf() closure.
+//
+void artistic_ior(color reflectivity, color edge_color, output color ior, output color extinction);
```

Comment:
Is the intent to provide the implementation of this function?

If left up to the renderer to provide, it would potentially make the `.oso` files non-portable.

Reply:
Also, why the output arg? Can we make the function return the result?

Reply:
Yes, this is the parameter remapping from Gulbrandsen (Artist Friendly Metallic Fresnel), and the implementation will be provided.

Reply:
About the return values, do you mean returning a struct, since there are two floats to return?

Reply:
Sorry, I didn't notice there were two. Looks good.

Reply:
I think you might as well supply the reflectivity and edge_color directly to the conductor BSDF, and let the implementation decide how it wants to convert these (which have well-defined meaning as reflected color at normal and near-grazing incidence) internally to the conductor IOR and k.

Reply:
I kinda agree with @portsmouth here.

Reply:
The `edge_color` here is not a real color, it's the 'g' parameter in Gulbrandsens parameterization, which has a strange indirect and nonlinear relationship to the final color. So I'm not sure we can make such a generalization and let the implementation choose if this is to be interpreted as f90 or 'g'.

We actually had this (Gulbrandsens parameterization) on the `conductor_bsdf` node in the first version of the MaterialX PBS library. But at Siggraph last year a lot of good points where raised about issues with this parameterization (mainly from Naty at the PBS course), so we decided to change it to use the physical eta/kappa parameters by default, and then anyone can remap that if desired as we do with the `artistic_ior` function.

Following Siggraph last year we started a discussion on this topic over at the Standard Surface github repo:
https://github.com/Autodesk/standard-surface/issues/12

On our side we haven't had the bandwidth to invest more time on that discussion yet (other than deciding to move away from using Gulbrandsens model by default). Maybe the Fresnel parameter discussion should continue in that other thread. Until we have resolved that I feel keeping the physical eta/kappa parameterization is maybe the safest bet for now.

Also, note that we already have the `generalized_schlick_bsdf` closure, with full artistic control over f0 and f90. This can be used for both dielectrics and metals if you prefer to have more control (with the risk of getting non-physical results)